llama_index/evaluation/retrieval/evaluator.py

```diff
@@ -23,25 +24,43 @@ class RetrieverEvaluator(BaseRetrievalEvaluator):
     Args:
         metrics (List[BaseRetrievalMetric]): Sequence of metrics to evaluate
         retriever: Retriever to evaluate.
+        node_postprocessors (Optional[List[BaseNodePostprocessor]]): Post-processor to apply after retrieval.
+
 
     """
 
     retriever: BaseRetriever = Field(..., description="Retriever to evaluate")
+    node_postprocessors: Optional[List[BaseNodePostprocessor]] = Field(
+        default=None, description="Optional post-processor"
+    )
 
     def __init__(
         self,
         metrics: Sequence[BaseRetrievalMetric],
         retriever: BaseRetriever,
+        node_postprocessors: Optional[List[BaseNodePostprocessor]] = None,
         **kwargs: Any,
     ) -> None:
         """Init params."""
         super().__init__(metrics=metrics, retriever=retriever, **kwargs)
```

Comment:
Actually, since this is a pydantic class, and node postprocessors is a field on the class, it should go into the super init rather than a self assign

Reply:
I am not super familiar with pydantic class yet, but can I confirm that it handle both of the following cases properly?
1. `node_postprocessors` is assigned through the constructor during the direct class instantiation.
2. `node_postprocessors` is passed through the class method `from_metric_names()` through `**kwargs`.

Reply:
Since it's a named argument in the init, it's not getting passed through kwargs.

While assigning to self works, it only works because `self.field = val` is happening after the init. If it was before, it would error out (this is a pydantic behavior)

So since it's a bit error prone, best to pass it purely through the super init 👍

Reply:
Thanks for clarifying. I have reflected that in my changes.

Reply:
Thanks!